pkg/cloudprovider/types.go

```diff
@@ -68,19 +68,28 @@ type NodeRequest struct {
 
 // InstanceType describes the properties of a potential node (either concrete attributes of an instance of this type
 // or supported options in the case of arrays)
-type InstanceType interface {
+type InstanceType struct {
```

Comment:
Make sure that callers start using the pointer version of this to avoid copying.